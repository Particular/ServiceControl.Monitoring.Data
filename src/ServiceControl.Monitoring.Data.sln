
Microsoft Visual Studio Solution File, Format Version 12.00
# Visual Studio 15
VisualStudioVersion = 15.0.26730.3
MinimumVisualStudioVersion = 10.0.40219.1
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ServiceControl.Monitoring.Data", "ServiceControl.Monitoring.Data\ServiceControl.Monitoring.Data.csproj", "{377C3A91-082F-4079-869E-C03546A9C58D}"
EndProject
Global
	GlobalSection(SolutionConfigurationPlatforms) = preSolution
		Debug|Any CPU = Debug|Any CPU
		Release|Any CPU = Release|Any CPU
	EndGlobalSection
	GlobalSection(ProjectConfigurationPlatforms) = postSolution
		{377C3A91-082F-4079-869E-C03546A9C58D}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{377C3A91-082F-4079-869E-C03546A9C58D}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{377C3A91-082F-4079-869E-C03546A9C58D}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{377C3A91-082F-4079-869E-C03546A9C58D}.Release|Any CPU.Build.0 = Release|Any CPU
	EndGlobalSection
	GlobalSection(SolutionProperties) = preSolution
		HideSolutionNode = FALSE
	EndGlobalSection
	GlobalSection(ExtensibilityGlobals) = postSolution
		SolutionGuid = {79E7301A-2805-475D-831D-3B8E9C8A5E6D}
	EndGlobalSection
EndGlobal
