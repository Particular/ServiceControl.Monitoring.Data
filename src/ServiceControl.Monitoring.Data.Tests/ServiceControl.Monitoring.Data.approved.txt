[assembly: System.Runtime.CompilerServices.InternalsVisibleToAttribute("ServiceControl.Monitoring.Data.Tests")]
[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.5.2", FrameworkDisplayName=".NET Framework 4.5.2")]

namespace ServiceControl.Monitoring.Data
{
    
    public class static LongValueWriterV1
    {
        public static void Write(System.IO.BinaryWriter outputWriter, System.ArraySegment<ServiceControl.Monitoring.Data.RingBuffer.Entry> chunk) { }
    }
    public class static OccurrenceWriterV1
    {
        public static void Write(System.IO.BinaryWriter outputWriter, System.ArraySegment<ServiceControl.Monitoring.Data.RingBuffer.Entry> chunk) { }
    }
    public class RawDataReporter : System.IDisposable
    {
        public RawDataReporter(System.Func<byte[], System.Threading.Tasks.Task> sender, ServiceControl.Monitoring.Data.RingBuffer buffer, ServiceControl.Monitoring.Data.WriteOutput outputWriter) { }
        public RawDataReporter(System.Func<byte[], System.Threading.Tasks.Task> sender, ServiceControl.Monitoring.Data.RingBuffer buffer, ServiceControl.Monitoring.Data.WriteOutput outputWriter, int flushSize, System.TimeSpan maxSpinningTime) { }
        public void Dispose() { }
        public void Start() { }
        public System.Threading.Tasks.Task Stop() { }
    }
    public class RingBuffer
    {
        public RingBuffer() { }
        public bool TryWrite(long value, int tag = 0) { }
        public struct Entry
        {
            public int Tag;
            public long Ticks;
            public long Value;
            public Entry(long ticks, long value, int tag = 0) { }
        }
    }
    public class TaggedLongValueWriterV1
    {
        public TaggedLongValueWriterV1() { }
        public int GetTagId(string tagName) { }
        public void Write(System.IO.BinaryWriter outputWriter, System.ArraySegment<ServiceControl.Monitoring.Data.RingBuffer.Entry> chunk) { }
    }
    public delegate void WriteOutput(System.ArraySegment<ServiceControl.Monitoring.Data.RingBuffer.Entry> entries, System.IO.BinaryWriter outputWriter);
}